### PR TITLE
Fix compilation issues for Helicopter and Sub

### PR DESCRIPTION
**Pull request by peterbarker**
This includes the fixes that were in #12731 

There's a minor behavioural change in toy mode - where I squashed a mode reason into a pre-existing one.

### CODE REVIEW

File: ArduCopter/ekf_check.cpp [hunk 1]
```diff
@@ -110,7 +110,14 @@ bool Copter::ekf_over_threshold()
     if (mag_variance.length() >= g.fs_ekf_thresh) {
         over_thresh_count++;
     }
-    if (!optflow.healthy() && (vel_variance >= (2.0f * g.fs_ekf_thresh))) {
+    bool vel_variance_bad = vel_variance >= (2.0f * g.fs_ekf_thresh);
```
Comment (rmackay9) [inline ArduCopter/ekf_check.cpp]: maybe instead we could have a variable called optflow_healthy which defaults to false and then set it to true if we've got optical flow included and healthy?

**Comment by rmackay9** [comment]
@bnsgeyer, can you check that you're happy with the "Copter: correct Heli compilation when some modes are disabled" commit in particular?

File: ArduCopter/ekf_check.cpp (hunk 1 above)
Comment (peterbarker) [reply ArduCopter/ekf_check.cpp]: Done.

**Comment by bnsgeyer** [comment]
@rmackay9  this appears to be the same changes @peterbarker had pinged me on in this PR #12243.   I had started making changes but never got back to it to ensure they were correct.  How should we proceed.  I'd prefer to continue working on the PR that Peter request I help him with.  I had made some changes and need to verify they are correct and provide the same response.

**Comment by peterbarker** [comment]
@bnsgeyer Where are your changes for the other PR?  I think I'd prefer to close that one in favour of this one, if that's OK.  I can incorporate any changes required into this PR.

**Comment by bnsgeyer** [comment]
I believe I made changes and force pushed them into your PR branch after squashing.  I'm looking now

**Comment by bnsgeyer** [comment]
@peterbarker just added another commit to #12243.  Please pull both commits in the old PR into your current PR.  I will test this tomorrow to ensure the behaviors are the same.

**Comment by peterbarker** [comment]
> @peterbarker just added another commit to #12243. Please pull both commits in the old PR into your current PR. I will test this tomorrow to ensure the behaviors are the same.

Done.

**Comment by rmackay9** [comment]
to be clear, I'm happy with the changes so once @bnsgeyer is happy I think it's ok to merge.

The PR affects Sub too but the changes are so small I can't imagine that @jaxxzer will mind.

**Comment by bnsgeyer** [comment]
I have done a little bit of testing.  I will do more tomorrow.  I think I might adjust one of the  conditions statements.  Sport really isn't manual throttle when it is on the ground.

**Comment by peterbarker** [comment]
On Sun, 3 Nov 2019, Bill Geyer wrote:

> I have done a little bit of testing. I will do more tomorrow. I think I might adjust one of the conditions statements. Sport really isn't manual throttle
> when it is on the ground.

OK, if you're good with behavioural changes like that I'm happy.  I don't 
think DRIFT is really manual throttle either, for the record :-)

**Comment by bnsgeyer** [comment]
@peterbarker, @lthall , and @rmackay9  I would argue that Drift is still a manual throttle mode.  It just has vertical speed feedback.  The pilot is still controlling throttle, it is just augmented slightly by vertical speed.   Not sure what else this affects but I would recommend changing the has_manual_throttle property for drift to true.

**Comment by peterbarker** [comment]
@bnsgeyer It looks like considering drift manual-throttle would correct a logging bug, at least.

I'd really rather not make such a change on *this* PR - which is only supposed to be fixing compilation bugs.

Without this PR going in we're not creating some binaries on our autotest server, which really isn't great.

Are you happy with this PR as-is?

**Comment by bnsgeyer** [comment]
@peterbarker I made a change to remove sport from the first conditions statement in should_use_landing_swash.   I am happy with this pr now.

**Comment by peterbarker** [comment]
Merged - thanks @bnsgeyer

**Pull request merged**